src/types.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+const (
+	CommitStatePending CommitState = "pending"
+	CommitStateSuccess CommitState = "success"
+	CommitStateFailure CommitState = "failure"
+	CommitStateError   CommitState = "error"
+
+	CheckSuiteActionCompleted ActionType = "completed"
+
+	IssueCommentActionCreated ActionType = "created"
+
+	CheckSuiteStatusRequested  CheckSuiteStatus = "requested"
+	CheckSuiteStatusInProgress CheckSuiteStatus = "in_progress"
+	CheckSuiteStatusCompleted  CheckSuiteStatus = "completed"
+
+	CheckSuiteConclusionSuccess        CheckSuiteConclusion = "success"
+	CheckSuiteConclusionFailure        CheckSuiteConclusion = "failure"
+	CheckSuiteConclusionNeutral        CheckSuiteConclusion = "neutral"
+	CheckSuiteConclusionCancelled      CheckSuiteConclusion = "cancelled"
+	CheckSuiteConclusionTimedOut       CheckSuiteConclusion = "timed_out"
+	CheckSuiteConclusionActionRequired CheckSuiteConclusion = "action_required"
+	CheckSuiteConclusionStale          CheckSuiteConclusion = "stale"
+)
+
+type ActionType string
+type CommitState string
+type CheckSuiteStatus string
+type CheckSuiteConclusion string
+
+type StatusBody struct {
+	State       CommitState `json:"state"`
+	Description string      `json:"description"`
+	Context     string      `json:"context"`
+	TargetUrl   string      `json:"target_url"`
+}
+
+type PullRequest struct {
+	Url         string `json:"url"`
```

Comment:
I'm not familiar with go but what is the `json:"ur"` telling here? Is that some sort of attribute that the json parser will read to map the value?

Reply:
Correct, it's the equivalent of the `JsonProperty` attribute. The go language spec supports arbitrary "struct tags" that packages can consume and use (see [examples](https://zchee.github.io/golang-wiki/Well-known-struct-tags/)).